jwk/option.go

```diff
@@ -30,10 +30,10 @@ var (
 	}
 )
 
-// Option represents a functional options pattern interface
+// Option represents a functional option
 type Option func(*jwkd) error
 
-// WithAthenzURL represents set athenzURL functional option
+// WithAthenzURL returns an AthenzURL functional option
 func WithAthenzURL(url string) Option {
```

Comment:
This is a sub-component.

https://github.com/yahoojapan/athenz-authorizer/blob/b56348ee1371d14e26ce2af5e4cc5226f548ea7d/authorizerd.go#L105-L154

`authorizerd` will pass the processed AthenzURL to the sub-components, so the sub-components do not need to do the same thing.

Reply:
@WindzCUHK No, I think you mentioned inconsistency problem on the issue. For your implementation I think it is also inconsistent and hard to maintain. Moreover when user want to use each daemon alone the logic become inconsistent. 
I suggest to move the scheme replacement code to authroizerd rather than in the option.

Reply:
@kevindiu 
I think it maybe hard to read if there are logic in both option and the constructor.

> Moreover when user want to use each daemon alone the logic become inconsistent.
Better to have the logic in all sub-components?

Reply:
I think we can
1. remove this logic on option.... 
2. copy this logic to all sub-component....
3. remove this logic...

I prefer 3 because I think this logic shouldn't be implemented at first...

Reply:
If we put the logic in the option I think it is harder to maintain...

Reply:
@kevindiu 
> I prefer 3 because I think this logic shouldn't be implemented at first...

Why do it even exits...😅

I prefer 2, and with the logic in option, as so option will throw error already.
https://github.com/yahoojapan/athenz-authorizer/blob/b56348ee1371d14e26ce2af5e4cc5226f548ea7d/jwk/option.go#L54-L56
And, we need this logic anyway as `https` is hard-cored.
We should check and throw error in the init process.
https://github.com/yahoojapan/athenz-authorizer/blob/b56348ee1371d14e26ce2af5e4cc5226f548ea7d/jwk/daemon.go#L121

Reply:
I am thinking we shouldn't append `https` on the prefix of the URL... And user should configure athenz URL correctly..

Reply:
> user should configure athenz URL correctly.

I think this is impossible. User is crazy.🤪

so...
1. fix option, throw error if NOT https or scheme is empty
2. fix everything, but sill need to append `https` scheme in someway for current interface to work
3. other idea...

Reply:
I disagree that users are crazy. 
For setting https here it loss the flexibility to change afterwards and also making the test code more complex.

However to support backward compatibility I think we can put the logic to all of the option classes.

Reply:
Or we should move this logic to authorization-proxy instead of implement in the library.

Reply:
solution?

1. change parent class only (this PR)
2. flatten the design, add logic to all classes later (another PR)